dipy/reconst/dti.py

```diff
@@ -2124,6 +2111,133 @@ def restore_fit_tensor(
         return params, extra
 
 
+def iterative_fit_tensor(design_matrix, data, *, jac=True,
+                         return_S0_hat=False,
+                         fit_type=None,
+                         num_iter=4,
+                         weights_method=None):
+    """ Iteratively Reweighted fitting for the DTI/DKI model.
+
+    Parameters
+    ----------
+
+    design_matrix : array of shape (g, ...)
+        Design matrix holding the covariants used to solve for the regression
+        coefficients.
+
+    data : array of shape ([X, Y, Z, n_directions], g)
+        Data or response variables holding the data. Note that the last
+        dimension should contain the data. It makes no copies of data.
+
+    return_S0_hat : bool
```

Comment:
`, optional` to add

Reply:
No blank lines needed between parameter docstrings. We are trying to move away from that so that docs are consistent. Applies to the whole PR.

Reply:
Okay, I will change for the stuff specific to my PR